Set Culture to en-US before deserializing

Thorjan opened the pull request:
When executing the GetJsonArray(string response) function the values of decimals/doubles are converted to the current culture format. If this culture has a ',' as decimal seperator the conversion at a later stage from JSON to object goes wrong.

The result is something like: "2850.5" is converted to "2850,5".
This happens in the following line of code "var dict = (Dictionary<string, object>)serializer.Deserialize<object>(response);"

Later on when you convert JSON to object, it will convert the "2850,5" value to "28505" because the JSON standard does not allow for a "," as decimal seperator.

Therefore before doing any deserialisation we should set the "CurrentCulture" to "en-US" in order to preserve the "." in decimals.

shtrip commented:
Thanks for submitting the pull request. Visual Studio is complaining it doesn't know **Thread** and **CultureInfo** so it doesn't compile. Probably one or two more 'using' statements are needed to import the necessary namespaces.

Thorjan commented:
This was indeed a mistake of mine. I copied the changes to the git repository and forgot to include the using statements.

using System.Globalization & using System.Threading were added.

antonrademaker commented:
I think it's better to use the CultureInfo.InvariantCulture property instead of a new CultureInfo("en-US")?

shtrip commented:
I Agree with @antonrademaker. InvariantCulture is the right thing to use here.

Thorjan commented:
That is indeed a better way.

I updated to Invariant Culture and also introduced the same mechanism in the "GetJsonObject" function since it also calls upon the "serializer.Deserialize" method

shtrip commented:
Tested the changes and merged.
Thanks for the fix!

Thorjan commented:
Glad I coulf be of of help...
